tools/memap.py

```diff
@@ -98,44 +145,44 @@ def check_new_section_gcc(self, line):
         else:
             return False         # everything else, means no change in section
 
-    
-    def path_object_to_module_name(self, txt):
-        """ Parse a path to object file to extract it's module and object data
+
+    def parse_object_name_gcc(self, line):
+        """ Parse a path to object file
 
         Positional arguments:
         txt - the path to parse the object and module name from
         """
 
-        txt = txt.replace('\\', '/')
-        rex_mbed_os_name = r'^.+mbed-os\/(.+)\/(.+\.o)$'
-        test_rex_mbed_os_name = re.match(rex_mbed_os_name, txt)
+        line = line.replace('\\', '/')
+        rex_mbed_os_name = r'^.+\/(.+\.o)$'
```

Comment:
Several things:
1) Regular expressions are normally referred to with `re` in python, not `rex`.
2) This should be a Class or global variable to indicate to the reader (again me in 2 months time) that this is a value that will not change.